test/Microsoft.TemplateEngine.Cli.UnitTests/ParserTests/TabCompletionTests.cs

```diff
@@ -0,0 +1,263 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.CommandLine;
+using System.CommandLine.Parsing;
+using FakeItEasy;
+using Microsoft.TemplateEngine.Abstractions;
+using Microsoft.TemplateEngine.Cli.Commands;
+using Microsoft.TemplateEngine.Edge;
+using Microsoft.TemplateEngine.Edge.Settings;
+using Microsoft.TemplateEngine.Mocks;
+using Microsoft.TemplateEngine.TestHelper;
+using Xunit;
+
+namespace Microsoft.TemplateEngine.Cli.UnitTests.ParserTests
+{
+    public partial class TabCompletionTests
+    {
+        [Fact]
+        public void Instantiate_CanSuggestTemplateOption_StartsWith()
+        {
+            ITemplateEngineHost host = TestHost.GetVirtualHost(additionalComponents: BuiltInTemplatePackagesProviderFactory.GetComponents(includeTestTemplates: false));
+            var myCommand = NewCommandFactory.Create("new", host, new TelemetryLogger(null, false), new NewCommandCallbacks());
+
+            var parseResult = myCommand.Parse("new console --framework net5.0 --l");
+            var suggestions = parseResult.GetSuggestions().ToArray();
+
+            Assert.Contains("--langVersion", suggestions);
+        }
+
+#pragma warning disable xUnit1004 // Test methods should not be skipped
+        [Fact(Skip = "not valid behavior for parser, should suggest --nuget-source")]
+#pragma warning restore xUnit1004 // Test methods should not be skipped
+        public void Install_GetSuggestionsAfterInteractive()
+        {
+            ITemplateEngineHost host = TestHost.GetVirtualHost(additionalComponents: BuiltInTemplatePackagesProviderFactory.GetComponents(includeTestTemplates: false));
+            var myCommand = NewCommandFactory.Create("new", host, new TelemetryLogger(null, false), new NewCommandCallbacks());
+
+            var parseResult = myCommand.Parse("new install --interactive ");
+            var result = parseResult.GetSuggestions().ToArray();
+
+            Assert.Contains("--nuget-source", result);
+        }
+
+        [Fact]
+        public void Install_GetSuggestionsAfterOptionWithoutArg()
+        {
+            ITemplateEngineHost host = TestHost.GetVirtualHost(additionalComponents: BuiltInTemplatePackagesProviderFactory.GetComponents(includeTestTemplates: false));
+            var myCommand = NewCommandFactory.Create("new", host, new TelemetryLogger(null, false), new NewCommandCallbacks());
+
+            var parseResult = myCommand.Parse("new install --nuget-source ");
+            var result = parseResult.GetSuggestions().ToArray();
+
+            Assert.Empty(result);
+        }
+
+#pragma warning disable xUnit1004 // Test methods should not be skipped
+        [Fact(Skip = "not valid behavior for parser, should suggest --interactive probably")]
```

Comment:
not expected behavior.
`--nuget-source` is `Option` that accepts multiple args, and one arg is already given.
Excepted that parser will suggest next option instead (`--interactive`), now nothing is returned

Reply:
https://github.com/dotnet/command-line-api/issues/1495